game-core/src/main/java/games/strategy/util/PointFileReaderWriter.java

```diff
@@ -40,30 +50,21 @@ private PointFileReaderWriter() {}
     checkNotNull(stream);
 
     final Map<String, Point> mapping = new HashMap<>();
-    try (Reader inputStreamReader = new InputStreamReader(new CloseShieldInputStream(stream), StandardCharsets.UTF_8);
-        LineNumberReader reader = new LineNumberReader(inputStreamReader)) {
-      @Nullable
-      String current = reader.readLine();
-      while (current != null) {
-        if (current.trim().length() != 0) {
-          readSingle(current, mapping);
-        }
-        current = reader.readLine();
-      }
-    }
+    readStream(stream, current -> readSingle(current, mapping));
     return mapping;
   }
 
-  private static void readSingle(final String line, final Map<String, Point> mapping) throws IOException {
-    final StringTokenizer tokens = new StringTokenizer(line, "", false);
-    final String name = tokens.nextToken("(").trim();
-    if (mapping.containsKey(name)) {
-      throw new IOException("name found twice:" + name);
+  private static void readSingle(final String line, final Map<String, Point> mapping) {
+    final Matcher matcher = singlePointPattern.matcher(line);
+    if (matcher.find()) {
+      final String name = matcher.group(1);
+      if (mapping.containsKey(name)) {
+        throw new IllegalArgumentException("name found twice:" + name);
+      }
+      mapping.put(name, new Point(Integer.parseInt(matcher.group(2)), Integer.parseInt(matcher.group(3))));
+    } else {
+      throw new IllegalArgumentException("Invalid Point Pattern");
```

Comment:
Can we add some info here so we know which point pattern was invalid?

Reply:
Good catch. I wanted to change that but I forgot 😅